ocaml/testsuite/tests/typing-local/regions.ml

```diff
@@ -18,6 +18,37 @@ let check_empty name =
 
 let () = check_empty "startup"
 
+let foo x = local_
+  ignore_local (Some x);
+  ()
+
+let foo_bad : 'a -> unit = Obj.magic foo
+
+external follow : int -> ('a [@local_opt]) = "%int_as_pointer"
+
+let[@inline never] int_as_pointer_local x =
+  foo_bad x;
+  (* The region should be preserved by the following call; hence no stack space
+     leaking *)
+  let _ = opaque_identity (follow 0) in
+  ()
+
+let[@inline never] int_as_pointer_global x =
+  (* The current function region will be eliminated, because the following two
+     function calls don't allocate on the region (superficially). The first call
+     secretly does, hence stack space leaking. *)
+  foo_bad x;
+  let _ = Sys.opaque_identity (follow 0) in
+  ()
+
+let () =
+  int_as_pointer_global 42;
+  check_empty "int_as_pointer (global)"
```

Comment:
I think you should create a `check_not_empty` that prints `OK` if the thing leaks and something like `Failed` if it doesn't. Otherwise the reference makes it look like the test is broken. You should be able to factor out the common part of `check_empty` and `check_not_empty` so that there isn't much new code.